Shared/Realm.Sync.Shared/SyncConfiguration.cs

```diff
@@ -20,14 +20,34 @@
 
 namespace Realms.Sync
 {
+    /// <summary>
+    /// An SyncConfiguration is used to setup a Realm that can be synchronized between devices using the Realm Object Server.
+    /// A valid <see cref="User"/> is required to create a SyncConfiguration.
```

Comment:
\<see cref="SyncConfiguration"\>, I guess? Or is it silly when describing itself?